Modelica/Clocked/BooleanSignals/Sampler/Utilities/AssignClockToSquareWaveHold.mo

```diff
@@ -14,7 +14,7 @@ protected
   Boolean y2(start=y_start);
 equation
   // dummy condition to relate clock of u with y2
-  y2 = if u == false then not previous(y2) else not previous(y2);
+  y2 = if not u then not previous(y2) else not previous(y2);
```

Comment:
```suggestion
  y2 = if u then not previous(y2) else not previous(y2);
```
Note that then and else-branches are identical, and it would be by good to relate clock in another way.